include/git2/branch.h

```diff
@@ -169,116 +166,127 @@ GIT_EXTERN(int) git_branch_lookup(
 	git_branch_t branch_type);
 
 /**
- * Return the name of the given local or remote branch.
+ * Get the branch name
  *
- * The name of the branch matches the definition of the name
- * for git_branch_lookup. That is, if the returned name is given
- * to git_branch_lookup() then the reference is returned that
- * was given to this function.
+ * Given a reference object, this will check that it really is a branch (ie.
+ * it lives under "refs/heads/" or "refs/remotes/"), and return the branch part
+ * of it.
  *
- * @param out where the pointer of branch name is stored;
- * this is valid as long as the ref is not freed.
- * @param ref the reference ideally pointing to a branch
+ * @param out Pointer to the underlying, truncated, reference name.
+ *        Owned by ref, do not free.
```

Comment:
Truncated sounds a bit weird here. I'm not a native speaker, but doesn't truncation indicate that we're chopping something off of the end?

Reply:
True, though I'm not as well, and I'm realizing I lack an "anti-synonym" for truncation 😆. This is actually the ref name, offset by its "namespace" length, so `Pointer to the underlying reference name, with some offset added.` or something ?

Reply:
What about "Pointer to the abbreviated reference name"? That would match gitglossary(7):

    ref
        A name that begins with refs/ (e.g.  refs/heads/master) that points
        to an object name or another ref (the latter is called a symbolic
        ref). For convenience, a ref can sometimes be abbreviated when used
        as an argument to a Git command; see gitrevisions(7) for details.
        Refs are stored in the repository.